dulwich/archive.py

```diff
@@ -1,93 +1,109 @@
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-# You should have received a copy of the licenses; if not, see
-# <http://www.gnu.org/licenses/> for a copy of the GNU General Public License
-# and <http://www.apache.org/licenses/LICENSE-2.0> for a copy of the Apache
-# License, Version 2.0.
-#
-
 """Generates tarballs for Git trees.
 
+archive.py -- Creating an archive from a tarball
+Copyright (C) 2015 Jonas Haag <jonas@lophus.org>
+Copyright (C) 2015 Jelmer Vernooij <jelmer@jelmer.uk>
+
+Dulwich is dual-licensed under the Apache License, Version 2.0 and the GNU
+General Public License as public by the Free Software Foundation; version 2.0
+or (at your option) any later version. You can redistribute it and/or
+modify it under the terms of either of these two licenses.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+You should have received a copy of the licenses; if not, see
+<http://www.gnu.org/licenses/> for a copy of the GNU General Public License
+and <http://www.apache.org/licenses/LICENSE-2.0> for a copy of the Apache
+License, Version 2.0.
+
 """
 
 import posixpath
+
 import stat
+
 import tarfile
+
 import struct
+
 from os import SEEK_END
+
 from io import BytesIO
+
 from contextlib import closing
 
 
-class ChunkedBytesIO(object):
+class ChunkedBytesIO(object):  # pylint: disable=too-few-public-methods
     """Turn a list of bytestrings into a file-like object.
 
     This is similar to creating a `BytesIO` from a concatenation of the
     bytestring list, but saves memory by NOT creating one giant bytestring
-    first::
+    first:
 
         BytesIO(b''.join(list_of_bytestrings)) =~= ChunkedBytesIO(
             list_of_bytestrings)
     """
 
     def __init__(self, contents):
+        """Init."""
         self.contents = contents
         self.pos = (0, 0)
 
     def read(self, maxbytes=None):
+        """Read.
+
+        Args:
+            maxbytes: int
+
+        Returns:
+            Bytestrings
+        """
         if maxbytes < 0:
             maxbytes = float("inf")
 
         buf = []
         chunk, cursor = self.pos
 
         while chunk < len(self.contents):
-            if maxbytes < len(self.contents[chunk]) - cursor:
-                buf.append(self.contents[chunk][cursor : cursor + maxbytes])
-                cursor += maxbytes
-                self.pos = (chunk, cursor)
-                break
-            else:
+            if maxbytes >= len(self.contents[chunk]) - cursor:
```

Comment:
You can just drop the else: statement, there's no need to change the rest of the code.

Reply:
I don't understand. Why invert the if statement?

Reply:
I think it’s because it’s easier to know why the loop is interrupted.

https://docs.quantifiedcode.com/python-anti-patterns/correctness/else_clause_on_loop_without_a_break_statement.html

```python3
        while chunk < len(self.contents):
            if maxbytes < len(self.contents[chunk]) - cursor:
                buf.append(self.contents[chunk][cursor : cursor + maxbytes])
                cursor += maxbytes
                self.pos = (chunk, cursor)
                break
            else:
                buf.append(self.contents[chunk][cursor:])
                maxbytes -= len(self.contents[chunk]) - cursor
                chunk += 1
                cursor = 0
                self.pos = (chunk, cursor)
```

to

```python3
        while chunk < len(self.contents):
            if maxbytes >= len(self.contents[chunk]) - cursor:
                buf.append(self.contents[chunk][cursor:])
                maxbytes -= len(self.contents[chunk]) - cursor
                chunk += 1
                cursor = 0
                self.pos = (chunk, cursor)
            else:
                buf.append(self.contents[chunk][cursor: cursor + maxbytes])
                cursor += maxbytes
                self.pos = (chunk, cursor)
                break
```

Reply:
https://pycodequ.al/docs/pylint-messages/r1723-no-else-break.html

Reply:
And it's just an example of `for` loop

The same can be applied to the `while` loop

Reply:
Yes, so I hope I can pass the check without adding any ignore rules